Projections/Filters/TokenizedContentTypeFilter.cs

```diff
@@ -29,16 +29,24 @@ public void ApplyFilter(FilterContext context)
         {
             var formValues = new TokenizedStringValueListFilterFormElements(context.State);
 
-            if (formValues.Values.Any()) context.Query = context.Query.ForType(formValues.Values);
+            if (formValues.Values.Any())
+            {
+                if (formValues.Matches) context.Query.ForType(formValues.Values);
+                else context.Query.Where(
+                    alias => alias.ContentItem(),
+                    filter => filter.Not(ex => ex.InG("ContentType.Name", formValues.Values)));

```

Comment:
@I3undy a concept nem volt rossz, viszont lehet egyszerűbben a fentieknek megfelelően.
Még annyi magyarázatot tennék hozzá, hogy a következő is működne:
```
context.Query.Where(
    alias => alias.Named("ct"),
    filter => filter.Not(ex => ex.InG("Name", formValues.Values)));
```
Ehhez viszont az kell, hogy a "ct" nevű alias már létezzen, tehát feltételezzük, hogy a query már bejoinolta a ContentTypeRecord táblát (aminek az alias-a mindig "ct", ld. DefaultHqlQuery.BindTypeCriteria). Ez egy viszonylag safe bet, mert szinte minden Query-t úgy kezdünk, hogy milyen type lehet benne, de nem lehetünk biztosak benne, hogy mindig így lesz (meg ez egy open source modul). A fenti megoldáshoz viszont nem kell, mert a record reference mentén hivatkozunk a content type-ra (amit az NHibernate old fel, szóval lényegében egy implicit join).